homeassistant/components/ecovacs_mqtt/config_flow.py

```diff
@@ -0,0 +1,271 @@
+                        )
+                    )
+                }
+            ),
+        )
+
+    async def async_step_auth(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the auth step."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                self._devices = await _retrieve_devices(self.hass, user_input)
+            except ClientError:
+                _LOGGER.debug("Cannot connect", exc_info=True)
+                errors["base"] = "cannot_connect"
+            except InvalidAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if not errors:
+                self._data.update(user_input)
+                if self._entry:
+                    # reauthentication
+                    self.hass.config_entries.async_update_entry(
+                        self._entry, data=self._data
+                    )
+                    self.hass.async_create_task(
+                        self.hass.config_entries.async_reload(self._entry.entry_id)
+                    )
+                    return self.async_abort(reason="reauth_successful")
+
+                self._async_abort_entries_match(
+                    {CONF_USERNAME: user_input[CONF_USERNAME]}
+                )
+
+                if len(self._devices) == 0:
+                    return self.async_abort(reason="no_supported_devices_found")
+
+                return await self.async_step_options()
+
+        return self.async_show_form(
+            step_id="auth",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_USERNAME): selector.TextSelector(
+                            selector.TextSelectorConfig(
+                                type=selector.TextSelectorType.TEXT
+                            )
+                        ),
+                        vol.Required(CONF_PASSWORD): selector.TextSelector(
+                            selector.TextSelectorConfig(
+                                type=selector.TextSelectorType.PASSWORD
+                            )
+                        ),
+                        vol.Required(CONF_COUNTRY): selector.CountrySelector(),
+                    }
+                ),
+                user_input or {CONF_COUNTRY: self.hass.config.country},
+            ),
+            errors=errors,
+        )
+
+    async def async_step_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the options step."""
+
+        errors = {}
+        if user_input is not None:
+            if len(user_input[CONF_DEVICES]) < 1:
+                errors[CONF_DEVICES] = "select_robots"
```

Comment:
Is there a reason we want the user to choose which devices they want to add? Usually the guideline is 1 config entry = 1 device or 1 config entry = 1 account.

Reply:
Currently it is `1 config entry = 1 account.`

Ecovacs allows to share your bot with another account and all bots will be returned.
My parent are sharing their bot with me, so I can control it when I'm at their home. But I don't want to add their bot on my home's instance.
For this reason a user can pick, which devices he wants to add it.

Another example is, if you have one account with multiple bots (one for each house) but each house as it's own HA instance.